### PR TITLE
Fixed crashing example in slicing section

### PR DESCRIPTION
**Pull request by schlupa**
The first example in the slicing section fails to build as it tries to link the external symbol foo which is not provided.

I also enhanced the other example to be a little bit more interesting for casual visitors. Examples which display only 'Succeed without output.' are completely useless.

### CODE REVIEW

File: spec/arrays.dd [hunk 1]
```diff
@@ -165,7 +165,10 @@ $(H2 $(LNAME2 slicing, Slicing))
 
 $(SPEC_RUNNABLE_EXAMPLE_COMPILE
```
Comment (wilzbach) [inline spec/arrays.dd]: Let's better use `SPEC_RUNNABLE_EXAMPLE_RUN` here, s.t. the spec tester actually runs the example?

File: spec/lex.dd [hunk 2]
```diff
@@ -600,7 +600,7 @@ $(GNAME DecimalInteger):
     $(GLINK NonZeroDigit) $(GLINK DecimalDigitsUS)
 
 $(GNAME BinaryInteger):
-    $(GLINK BinPrefix) $(GLINK BinaryDigitsUS)
+    $(GLINK BinPrefix) $(GLINK BinaryDigitsNoSingleUS)
```
Comment (ZombineDev) [inline spec/lex.dd]: @thewilsonator this change looks completely unrelated.

Comment (thewilsonator) [reply spec/lex.dd]: Ohh, good catch.

Comment (thewilsonator) [reply spec/lex.dd]: Hmm i think I must have `git reset --soft HEAD~N` for `N` equals one more than I thought. I was fighting both `git` and the stupid whitespace checker a lot with this, that that happened is not really all that surprising.

Comment (ZombineDev) [reply spec/lex.dd]: When you do changes to people's own branches be sure to make backups and then compare your changes with the backed up branch before force pushing.
